docs/html/user_guide.rst

```diff
@@ -861,6 +861,296 @@ of ability. Some examples that you could consider include:
+      werkzeug==2.0.2 --hash sha256:63d3dc1cf60e7b7e35e97fa9861f7397283b75d765afcaefd993d6046899de8f
+      wheel==0.37.1 --hash sha256:4bdcd7d840138086126cd09254dc6195fb4fc6f01c050a1d7236f2630db1d22a
+      cachetools==4.2.4 --hash sha256:92971d3cb7d2a97efff7c7bb1657f21a8f5fb309a37530537c71b1774189f2d1
+      certifi==2021.10.8 --hash sha256:d62a0163eb4c2344ac042ab2bdf75399a71a2d8c7d47eac2e2ee91b9d6339569
+      charset-normalizer==2.0.10 --hash sha256:cb957888737fc0bbcd78e3df769addb41fd1ff8cf950dc9e7ad7793f1bf44455
+      idna==3.3 --hash sha256:84d9dd047ffa80596e0f246e2eab0b391788b0503584e8945f2368256d2735ff
+      pyasn1-modules==0.2.8 --hash sha256:a50b808ffeb97cb3601dd25981f6b016cbb3d31fbf57a8b8a87428e6158d0c74
+      requests-oauthlib==1.3.0 --hash sha256:7f71572defaecd16372f9006f33c2ec8c077c3cfa6f5911a9a90202beb513f3d
+      rsa==4.8 --hash sha256:95c5d300c4e879ee69708c428ba566c59478fd653cc3a22243eeb8ed846950bb
+      six==1.16.0 --hash sha256:8abb2f1d86890a2dfb989f9a77cfcfd3e47c2a354b01111771326f8aa26e0254
+      urllib3==1.26.7 --hash sha256:c4fdf4019605b6e5423637e01bc9fe4daef873709a7973e195ceba0a62bbc844
+      oauthlib==3.1.1 --hash sha256:42bf6354c2ed8c6acb54d971fce6f88193d97297e18602a3a886603f9d7730cc
+      pyasn1==0.4.8 --hash sha256:39c7e2ec30515947ff4e87fb6f456dfc6e84857d34be479c9d4a4ba4bf46aa5d
+      setuptools==60.3.1 --hash sha256:2932bfeb248c648dc411ea9714d5a6de7a33ef1a0db2f0fce644d8172b0479e8
+
+JSON schema
+-----------
+
+The JSON report is described below with fields given fake mypy type annotations corresponding to the output of `json.load() <https://docs.python.org/3/library/json.html#json.load>`_ upon reading the JSON report. The report contains multiple top-level fields:
+
+* ``experimental: bool``: set to ``True`` as the format is not yet stable.
+* ``input_requirements: List[str]``: strings describing the requirements provided to the pip resolver. Can be parsed with `packaging.requirements.Requirement <https://packaging.pypa.io/en/latest/requirements.html>`_.
+* ``python_version: str``: a string describing the python interpreter version the resolve was performed for. Can be parsed with `packaging.specifiers.SpecifierSet <https://packaging.pypa.io/en/latest/specifiers.html>`_. Currently always an exact ``==`` constraint.
+* ``candidates: Dict``: each package that would have been downloaded with ``pip download`` is represented in the ``candidates`` dict. Each key is the ``name`` of a requirement, since each dependency package name is satisfied by exactly one candidate in the final resolve.
+
+Candidates
+----------
+
+Each element of the ``candidates`` dict has the following fields:
+
+* ``requirement: str``: an ``==`` requirement string for the exact version of the candidate that would have been fetched by ``pip download``. Can be parsed with `packaging.requirements.Requirement <https://packaging.pypa.io/en/latest/requirements.html>`_.
```

Comment:
What if the provided requirement was a direct url ? 

Here my suggestion would be to provide this field only if it was a top level requirement. I.e. user provided via the pip command line or requirement file. I'd keep here the original requirement line as close as possible to what the user provided.

The absence of this field would mean that the package was installed as a dependency.

Additional information (e.g. version downloaded, etc) can be found in the `metadata` field.

Reply:
I think that totally makes sense, especially in light of @sschuberth's note that having the concrete version of the package outside of this requirement string would be useful: https://github.com/pypa/pip/pull/10748#issuecomment-1095120205.

> The absence of this field would mean that the package was installed as a dependency.

I really like this. Instead of absence, however, we could maybe provide the package that had it as a dependency?

Reply:
In #10771 I currently have a `requested` flag that echoes the [REQUESTED metadata from PEP 376](https://peps.python.org/pep-0376/#requested).

> Instead of absence, however, we could maybe provide the package that had it as a dependency?

Having the actual packages that require another one could be a useful shortcut. There might be more than one, though.
This information can also be reconstructed by looking at the `requires_dist` metadata fields, so my first approach is to leave that out of the first version, with the goal of getting a MVP out and then build on it.